### ISSUE TITLE
Collabora not opening MSoffice documents

### ISSUE THREAD

**Issue by thorsten-schwartz** [nextcloud/android#11284]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

If trying to open a MSoffice document within Android nextcloud client it won't open. After about 20 secondary a message appears time too long.
Within the web GUI in Firefox Collabora works as expected.
I have Collabora on my server inside docker. 

### Expected behaviour

The office document docx should open and be able to be used. 

### Actual behaviour

See above. The document is not opening.
![Screenshot_20230122_143744_com nextcloud client](https://user-images.githubusercontent.com/54764492/213919532-c2c80f35-defc-4bbd-ada5-86ed9db526ec.jpg)

### Android version

10

### Device brand and model

Huawei P20 Pro

### Stock or custom OS?

Stock

### Nextcloud android app version

3.23.1

### Nextcloud server version

22

### Using a reverse proxy?

Yes

**Comment by thorsten-schwartz** [comment]
Android is V11 called EMUI12 at Huawei. But same error on other mobile clients on Android.

**Comment by joshtrichards** [comment]
Are you still experiencing this issue?

I can't reproduce this, but there are a number of differences between my environment and yours - plus unknowns about your environment. In addition, I note you're running v22 NC which is end-of-life.

If you're still having this issue: 

1. What edition and version of Collabora Online?
2. Did this work for you previously? (I realize this should work, but trying to get an idea of when it may have stopped working for you)
3. Have you submitted a possible bug report over at https://github.com/CollaboraOnline/online ?

**Comment by thorsten-schwartz** [comment]
Hi Josh,

I have still the same behaviour!

I updated to NC 25.0.4,  Nextcloud Office 7.1.2 and Collabora Docker Image from today (19.03.23)

No and this never worked before.
I didn't tell Collabrora till today.

Greets and thanks for your help.
Thorsten

**Comment by christianlupus** [comment]
Hello. I am most probably affected by this as well.

I am running collabora as a dedicated docker container (not included in any NC container). I am using the CODE version `22.05.12.4 git hash: bcbca64 (E)`.

I have two servers attached, one with NC 26.0.0 and one with NC 23.0.3 (I know, EOL). Both show the effect that on desktop, the Collabora GUI loads, on the Android client it does not.

---

Investigating further, I checked the network connections made to both the server (only the NC26) and to Collabora.

From desktop these requests are made (recorded in the reverse proxy):

```
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:06 +0200] "PUT /ocs/v2.php/apps/user_status/api/v1/heartbeat?format=json HTTP/1.1" 200 155 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:07 +0200] "PROPFIND /remote.php/dav/files/christianwolf/test.odt HTTP/1.1" 207 529 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:11 +0200] "GET /index.php/apps/richdocuments/index?fileId=4911&requesttoken=FOO&path=%2Ftest.odt HTTP/1.1" 200 11169 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:14 +0200] "GET /ocs/v2.php/apps/user_status/api/v1/user_status?format=json HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:14 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications?format=json HTTP/1.1" 304 - "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
collabora.example.com 84.147.31.75 - - [28/Mar/2023:09:33:17 +0200] "POST /browser/2d5b312/cool.html?WOPISrc=https%3A%2F%2Fcloud.example.com%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2Ffoo&title=test.odt&lang=de&closebutton=1&revisionhistory=1 HTTP/1.1" 200 31451 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:21 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:32 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 256 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:42 +0200] "GET /index.php/avatar//64 HTTP/1.1" 404 13067 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:45 +0200] "GET /index.php/apps/files/api/v1/stats HTTP/1.1" 200 165 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:50 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:34:02 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 256 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
collabora.example.com 84.147.31.75 - - [28/Mar/2023:09:33:21 +0200] "GET /cool/https%3A%2F%2Fcloud.example.com%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2Ffoo%3Faccess_token%3DBAR%26access_token_ttl%3D1680024791000/ws?WOPISrc=https%3A%2F%2Fcloud.example.com%2Findex.php%2Fapps%2Frichdocuments%2Fwopi%2Ffiles%2Ffoo&compat=/ws HTTP/1.1" 101 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
```

From the mobile app, these requests were logged:

```
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:31:59 +0200] "GET /ocs/v2.php/cloud/capabilities?format=json HTTP/1.1" 200 8865 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:00 +0200] "GET /ocs/v2.php/apps/external/api/v1?format=json HTTP/1.1" 200 241 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:31:59 +0200] "POST /ocs/v2.php/apps/richdocuments/api/v1/document?format=json HTTP/1.1" 200 214 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:31:59 +0200] "GET /ocs/v2.php/cloud/user?format=json HTTP/1.1" 200 603 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:00 +0200] "GET /index.php/apps/richdocuments/direct/O8Qtc HTTP/1.1" 200 7316 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:01 +0200] "GET /index.php/core/js/oc.js?v=438381d6 HTTP/1.1" 200 4795 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:03 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 256 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:02 +0200] "GET /dist/core-common.js?v=438381d6-9 HTTP/1.1" 200 15071827 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:14 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications?format=json HTTP/1.1" 304 - "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:14 +0200] "GET /ocs/v2.php/apps/user_status/api/v1/user_status?format=json HTTP/1.1" 200 155 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:19 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:31 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 256 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:45 +0200] "GET /index.php/apps/files/api/v1/stats HTTP/1.1" 200 165 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:49 +0200] "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/1.1" 304 - "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:56 +0200] "GET /status.php HTTP/1.1" 200 171 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:56 +0200] "GET /ocs/v2.php/cloud/capabilities?format=json HTTP/1.1" 304 - "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:55 +0200] "GET /ocs/v2.php/cloud/user?format=json HTTP/1.1" 200 603 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:57 +0200] "GET /ocs/v2.php/apps/user_status/api/v1/predefined_statuses?format=json HTTP/1.1" 200 288 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:57 +0200] "GET /ocs/v2.php/cloud/user?format=json HTTP/1.1" 200 603 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:58 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 1184 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:59 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 38467 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:32:59 +0200] "GET /ocs/v2.php/apps/files_sharing/api/v1/shares?path=%2F&reshares=true&subfiles=true HTTP/1.1" 200 1383 "-" "Mozilla/5.0 (Android) Nextcloud-android/3.24.1"
cloud.example.com 84.147.31.75 - - [28/Mar/2023:09:33:01 +0200] "PROPFIND /remote.php/dav/files/christianwolf/ HTTP/1.1" 207 256 "-" "Mozilla/5.0 (Linux) mirall/3.7.3git (Nextcloud, arch-6.2.1-arch1-1 ClientArchitecture: x86_64 OsArchitecture: x86_64)"
```

One can see multiple things:

The `POST` request to the collabora server is never made on the mobile for whatever reason. So, it seems not only a timing problem (loading took too long) but a bug.

Next, there is a difference in the requests to richdocuments: The desktop requests `/index.php/apps/richdocuments/index`, while the mobile requests `/index.php/apps/richdocuments/direct`. Does this make a significant difference?

I am unsure, if this bug is in the mobile app, the richdocuments app, or the CODE version. Due to the difference Collabora versions, I doubt this is caused solely by the Collabora release.

**Comment by tobiasKaminsky** [comment]
Probably this https://github.com/nextcloud/richdocuments/issues/2873 ?

**Comment by christianlupus** [comment]
Yes, that sounds familiar. I updated to NC 26 recently.

Should I cross-post, @tobiasKaminsky?

**Comment by Happyfeet01** [comment]
@tobiasKaminsky I solved it on the desktop by not running Collbora via Docker container but installing it directly on the host. Since then, Collabora Office loads quickly. In the Android app of Nextcloud still the error that the loading takes too long.

**Comment by joshtrichards** [comment]
Fixed in nextcloud/richdocuments#2874

**Comment by dorpauli** [comment]
I encounter the same problem again. Every office document is in endless load screen on mobile app. Multiple devices testet. Also OpenOffice documents aren't loading. I do not find any logs, that point to the problem. Everything up to date. Mobile app 3.27.0 Nextcloud Server 27.1.5 NC Office 8.2.4.

**Comment by joshtrichards** [comment]
@dorpauli I suspect you encountered #12338. It's fixed in #12349 / #12350. Should show up in 3.27.1 when released.
